File: backend/src/ytforge/interfaces/api/schemas/scripts.py
from __future__ import annotations

import uuid
from typing import Any

from pydantic import BaseModel, ConfigDict, Field

from ytforge.domain.enums import FactCheckVerdict, ScriptStatus


class ScriptCreateRequest(BaseModel):
    sections: dict[str, Any] = Field(default_factory=dict)
    model_used: str | None = None
    token_count: int | None = None


class ScriptStatusUpdateRequest(BaseModel):
    status: ScriptStatus


class ScriptRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    version: int
    status: ScriptStatus
    sections: dict[str, Any]
    model_used: str | None
    token_count: int | None


class FactCheckCreateRequest(BaseModel):
    verdict: FactCheckVerdict
    flags: list[Any] = Field(default_factory=list)
    model_used: str | None = None


class FactCheckRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    script_id: uuid.UUID
    script_version: int
    verdict: FactCheckVerdict
    flags: list[Any]
    model_used: str | None

File: frontend/src/lib/api/schemas/scripts.ts
import { z } from "zod";
import { FactCheckVerdict, ScriptStatus } from "@/types/enums";

const scriptStatusSchema = z.enum([
  ScriptStatus.DRAFT,
  ScriptStatus.IN_REVIEW,
  ScriptStatus.APPROVED,
  ScriptStatus.REJECTED,
]);

const factCheckVerdictSchema = z.enum([FactCheckVerdict.PASSED, FactCheckVerdict.FLAGGED]);

export const scriptReadSchema = z.object({
  id: z.string().uuid(),
  project_id: z.string().uuid(),
  version: z.number(),
  status: scriptStatusSchema,
  sections: z.record(z.string(), z.unknown()),
  model_used: z.string().nullable(),
  token_count: z.number().nullable(),
});
export type ScriptRead = z.infer<typeof scriptReadSchema>;

export const factCheckReadSchema = z.object({
  id: z.string().uuid(),
  script_id: z.string().uuid(),
  script_version: z.number(),
  verdict: factCheckVerdictSchema,
  flags: z.array(z.unknown()),
  model_used: z.string().nullable(),
});
export type FactCheckRead = z.infer<typeof factCheckReadSchema>;

export type ScriptCreateRequest = {
  sections?: Record<string, unknown>;
  model_used?: string | null;
  token_count?: number | null;
};

export type ScriptStatusUpdateRequest = {
  status: ScriptStatus;
};

export type FactCheckCreateRequest = {
  verdict: FactCheckVerdict;
  flags?: unknown[];
  model_used?: string | null;
};
